Formula/mesa.rb

```diff
@@ -47,6 +47,7 @@ class Mesa < Formula
     depends_on "libxv"
     depends_on "libxvmc"
     depends_on "libxxf86vm"
+    depends_on "llvm" # can't use from macOS, it isn't found
```

Comment:
That doesn't make sense.
```ruby
uses_from_macos "foo"
```
is the exact same thing as
```ruby
on_linux do
  depends_on "foo"
end
```

Reply:
`uses_from_macos "llvm"` is a false statement, it isn't found and isn't used.

Reply:
`uses_from_macos "llvm"` is a no-op on macOS (because of the equivalence above), so it being false doesn't really make sense.

Reply:
It may be equivalent in case formula actually detects and uses stock `llvm`, it's not the case for `mesa`, so I wanted to be more correct in stating `llvm` dependency in only `on_linux`.

Reply:
So what does it use on macOS?

Reply:
It uses stock `llvm`, but only as any other compiler. `mesa` is tied with `llvm` on other platforms, where real drivers work, e.g. `llvm` has backend for Radeon GPUs.

Reply:
I don't get it, if it uses `llvm` on macOS the `uses_from_macos "llvm"` is correct right?

Reply:
AFAIK, we don't set `uses_from_macos "llvm"` for all formula that just needs any C compiler.

Reply:
I mean that `mesa` may be compiled with any modern compiler and it doesn't use any LLVM-specific features macOS, because it doesn't recognize it as suitable LLVM instance in case of stock macOS.
And in case of advertising brewed LLVM it detects and "uses" it, but the resulting product is the same.

Reply:
So what is the solution? It doesn't need LLVM on macOS, only on Linux, so let's keep current code?

Reply:
Done